modifiers.c

```diff
@@ -0,0 +1,111 @@
+#include <inttypes.h>
+#include <stdbool.h>
+#include <stdio.h>
+
+#include <drm_fourcc.h>
+
+#include "tables.h"
+
+static const char *amd_tile_version_str(uint64_t tile_version) {
+	switch (tile_version) {
+	case AMD_FMT_MOD_TILE_VER_GFX9:
+		return "GFX9";
+	case AMD_FMT_MOD_TILE_VER_GFX10:
+		return "GFX10";
+	case AMD_FMT_MOD_TILE_VER_GFX10_RBPLUS:
+		return "GFX10_RBPLUS";
+	}
+	return "Unknown";
+}
+
+static const char *amd_tile_str(uint64_t tile, uint64_t tile_version) {
+	switch (tile_version) {
+	case AMD_FMT_MOD_TILE_VER_GFX10_RBPLUS:
+		/* fallthrough */
+	case AMD_FMT_MOD_TILE_VER_GFX10:
+		/* fallthrough */
+	case AMD_FMT_MOD_TILE_VER_GFX9:
+		switch (tile) {
+		case AMD_FMT_MOD_TILE_GFX9_64K_S:
+			return "GFX9_64K_S";
+		case AMD_FMT_MOD_TILE_GFX9_64K_D:
+			return "GFX9_64K_D";
+		case AMD_FMT_MOD_TILE_GFX9_64K_S_X:
+			return "GFX9_64K_S_X";
+		case AMD_FMT_MOD_TILE_GFX9_64K_D_X:
+			return "GFX9_64K_D_X";
+		case AMD_FMT_MOD_TILE_GFX9_64K_R_X:
+			return "GFX9_64K_R_X";
+		}
+	}
+	return "Unknown";
+}
+
+static bool amd_gfx9_tile_is_x_t(uint64_t tile) {
+	switch (tile) {
+	case AMD_FMT_MOD_TILE_GFX9_64K_S_X:
+	case AMD_FMT_MOD_TILE_GFX9_64K_D_X:
+	case AMD_FMT_MOD_TILE_GFX9_64K_R_X:
+		return true;
+	}
+	return false;
+}
+
+static void print_amd_modifier(uint64_t mod) {
+	printf("AMD(");
+
+	uint64_t tile_version = AMD_FMT_MOD_GET(TILE_VERSION, mod);
+	uint64_t tile = AMD_FMT_MOD_GET(TILE, mod);
+	uint64_t dcc = AMD_FMT_MOD_GET(DCC, mod);
+	uint64_t dcc_retile = AMD_FMT_MOD_GET(DCC_RETILE, mod);
+
+	printf("TILE_VERSION = %s", amd_tile_version_str(tile_version));
+	printf(", TILE = %s", amd_tile_str(tile, tile_version));
+
+	if (dcc) {
+		printf(", DCC");
```

Comment:
Maybe print "DCC_RETILE" if it is set? I think you don't print it anywhere?

Reply:
Oh, right, this is missing. Fixed!